components/automate-deployment/pkg/manifest/client/http.go

```diff
@@ -109,6 +169,31 @@ func (c *HTTP) manifestFromURL(ctx context.Context, url string) (*manifest.A2, e
 		return nil, err
 	}
 
+	if !c.noVerify {
+		signatureURL := fmt.Sprintf("%s.asc", url)
+		logrus.WithField("url", signatureURL).Debug("Checking manifest signature")
+		sigReq, err := http.NewRequest("GET", signatureURL, nil)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to GET manifest signature")
+		}
+		sigReq = sigReq.WithContext(ctx)
+
+		sigResp, err := c.HTTPClient.Do(sigReq)
+		if err != nil {
+			return nil, err
+		}
+		defer sigResp.Body.Close() // nolint: errcheck
+
+		if sigResp.StatusCode != http.StatusOK {
+			return nil, errors.Errorf("Failed to GET manifest signature. status=%s", sigResp.Status)
+		}
+
+		_, err = openpgp.CheckArmoredDetachedSignature(packagesChefIOKeyRing, bytes.NewBuffer(body), sigResp.Body)
```

Comment:
I wonder if we want to read in the body ourselves first so we can differentiate between network errors when reading the request and actual verification failures more easily.  Hopefully the errors out of this library are clear though.